### PR TITLE
cleanup: delete catsrc and operatorgroup even if subscription is not found

### LINKED ISSUE #4085: Cleanup command doesn't cleanup from a failed run
## Bug Report

/cc @raffaelespazzoli 

#### What did you do?

Tried to use the `cleanup` command to cleanup from a failed `run` command.

```
$ operator-sdk run bundle --install-mode AllNamespaces quay.io/$repo/volume-expander-operator-controller-bundle:latest
FATA[0007] Failed to run bundle: create catalog: error creating catalog source: catalogsources.operators.coreos.com "volume-expander-operator-catalog" already exists
$ operator-sdk cleanup volume-expander-operator
FATA[0000] Uninstall operator: operator package "volume-expander-operator" not found
```

#### What did you expect to see?

Successful cleanup of any resources created by the run command, such that `operator-sdk run bundle` does not fail with an error about existing resources.

#### What did you see instead? Under which circumstances?

```
FATA[0000] Uninstall operator: operator package "volume-expander-operator" not found
```

#### Environment

**Operator type:**

N/A

**Kubernetes cluster type:**

N/A

`$ operator-sdk version`

1.1.0

`$ go version` (if language is Go)

N/A

`$ kubectl version`

N/A

### PR DESCRIPTION
**Pull request by joelanford**
**Description of the change:**

Updates the `operator-sdk cleanup` subcommand to cleanup the catalog source and operator group even when a subscription for the operator does not exist.

**Motivation for the change:**

In the event that a user runs `operator-sdk run` and an error occurs after the operator group and catalog source are created, but before a subscription is created, the operator group and catalog source must be manually cleaned up before retrying the `operator-sdk run` command.

In 1.1.0, the `operator-sdk cleanup` command will not cleanup the operator group and catalog source in this scenario.

This PR updates the `operator-sdk cleanup` command so that even when a subscription does not exist, it looks for the expected catalog source and operator group, and deletes them if they exist.

Closes #4085 

**Checklist**

If the pull request includes user-facing changes, extra documentation is required:
- [x] Add a new changelog fragment in `changelog/fragments` (see [`changelog/fragments/00-template.yaml`](https://github.com/operator-framework/operator-sdk/tree/master/changelog/fragments/00-template.yaml))
- [ ] ~Add or update relevant sections of the docs website in [`website/content/en/docs`](https://github.com/operator-framework/operator-sdk/tree/master/website/content/en/docs)~

### CODE REVIEW

File: changelog/fragments/fix-cleanup.yaml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+entries:
+  - description: >
+      Fixed an issue in `operator-sdk cleanup` that caused catalog
+      source and operator group objects not to be cleaned up if a
+      previous `operator-sdk run` command failed.
```
Comment (estroz) [inline changelog/fragments/fix-cleanup.yaml:5]: For clarity
```suggestion
      Fixed an issue in `operator-sdk cleanup` that caused `CatalogSource`
      and `OperatorGroup` objects not to be cleaned up if a
      previous `operator-sdk run` command failed.
```

**Review by estroz — APPROVED** [review]
/lgtm

**Review by camilamacedo86 — APPROVED** [review]
/lgtm

**Pull request merged**